src/parser.rs

```diff
@@ -2401,15 +2524,23 @@ impl Parser {
     }
 
     pub fn peek(&mut self) -> Option<Token> {
+        self.peek_bareword(NameStrictness::Strict)
```

Comment:
Not sure if it makes sense for the overall `peek()` which is poorly defined what it does right now, to just follow strict bareword semantics.

Reply:
I chose the strict to be the default because otherwise the lexing would break, for example `{ a: b }` would be lexed with `a:` as a single token with completely loosened bareword strictness.

Reply:
Would that still yield the other tokens like `:` just with bareword segmentation/lexing assumptions?

Reply:
The current `peek()` follows the same peek functionality as before, i.e., `:` is segmented as a separate token. Strict barewords are also the same barewords as before, i.e., only letters with `_`.